### PR TITLE
Logs Panel: Table UI - Misc UI tweaks

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
A collection of UI changes after discussion with squad and UI review.

1. Adds title tag to log density calculation
image

2. Breaks active and inactive labels into separate sections in the sidebar
3. Add the ability to reset active filters 
image

5. Clean/optimize sort code
6. Removing horizontal scroll of long label names, hiding overflow and adding title tag to label name. 
image

7. Switching order of items in toggle.
image

8. Unifying the titles across logs/table visualization
image
image

**Who is this feature for?**
Users of logs table in explore

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

**Who is this feature for?**
Users of logs tabexplore

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/Logs/LogsTableNavColumn.tsx [hunk 1]
```diff
@@ -51,96 +53,51 @@ function getStyles(theme: GrafanaTheme2) {
   };
 }
 
+const collator = new Intl.Collator('en', { sensitivity: 'base' });
```
Comment (matyax) [inline public/app/features/explore/Logs/LogsTableNavColumn.tsx:56]: Very nice.

**Review by matyax — COMMENTED** [review]
Looks good! Only one thing, I'm not sure if this is the expected behavior:

**Review by ivanahuckova — COMMENTED** [review]
I have some thoughts on some of these changes: 

> Breaks active and inactive labels into separate sections in the sidebar

I am not sure this change brings value as it creates "jumpy" title + we are losing valuable vertical space. We already have this information from checkboxes + moving selected items to the top. 

> Add the ability to clear active filters

The behaviour is slightly weird for me, cause it says `Clear all` but when you click on it, it doesn't clear all, it just removes selected filters and chooses time and message. So if we are using `Clear all`, I would really clear all fields. Or then replace `Clear all` with something else. 

https://github.com/grafana/grafana/assets/30407135/8d22bc8e-331e-4bd7-88ad-4102fc4887f2

> Renaming "Logs" to "Lines"

This switch is confusing for me because we use both the Logs panel and Table panel, and this toggle switches between them. So, why did we change the name to "Lines"? We can see lines in both visualizations—the table panel and the logs panel.

**Comment by gtk-grafana** [comment]
> Breaks active and inactive labels into separate sections in the sidebar

While this does use a bit more space for the extra header, without this change whenever the user selects something that is down the list, it will automatically jump/scroll to the top where the item you just interacted with is moved to. Also users were confused by the sort order when all items were in a single section. 

> Add the ability to clear active filters
I guess we could change it say "reset" instead of clear?

> Renaming "Logs" to "Lines"
I think we went for this because "Logs" also applies to the table view.

@niat22 can you review the above and let us know what you think?

**Comment by svennergr** [comment]
> While this does use a bit more space for the extra header, without this change whenever the user selects something that is down the list, it will automatically jump/scroll to the top where the item you just interacted with is moved to. Also users were confused by the sort order when all items were in a single section.

I was wondering if we even need to change ordering once a field is selected? Personally I also don't really like the additional section just for separation.

**Comment by ivanahuckova** [comment]
> I was wondering if we even need to change ordering once a field is selected?

That's a valid point. I'm wondering if changing the order occurs in other places in Grafana when you select or deselect something. For example, in log details and transformations, it doesn't seem to happen.

https://github.com/grafana/grafana/assets/30407135/a0b2694e-be95-44ae-bc1d-b781fe6e44f3

**Comment by gtk-grafana** [comment]
Some good points here. I'm on the fence, the current and proposed UI have advantages and disadvantages
Since this was a request from Niat/UX squad, should we merge this in as is and re-address in a later PR, or should we hold off and wait for Niat to get back from vacation?

File: public/app/features/explore/Logs/LogsTableNavColumn.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/features/explore/Logs/LogsTableNavColumn.tsx:56]: Should we use `en` in here? Other places in Grafana use `undefined`. So I am wondering if we should be consistent or if there is a specific reason to use this. 

image

**Review by ivanahuckova — COMMENTED** [review]
The search is broken when you remove searched value:

https://github.com/grafana/grafana/assets/30407135/e44b3fc3-1591-4c6f-b6f8-aa6248790efd

**Comment by gtk-grafana** [comment]
@ivanahuckova good catches, the debounce was causing a race condition so I removed it as it was a carry over from the prometheus usage of ufuzzy, prom can have thousands to millions of metrics or labels to search through. 
Also changed en -> undefined to match other usages in Grafana.

**Review by ivanahuckova — APPROVED** [review]
LGTM! 🚀

**Pull request merged**